tendermint/src/consensus/state.rs

```diff
@@ -68,33 +64,34 @@ impl PartialOrd for State {
 mod tests {
     use super::State;
     use crate::block;
+    use std::convert::TryFrom;
 
     #[test]
     fn state_ord_test() {
         let new = State {
-            height: block::Height::from(9001u64),
-            round: 0,
+            height: block::Height::try_from(9001u64).unwrap(),
+            round: block::Round::try_from(0).unwrap(),
             step: 0,
             block_id: None,
         };
 
         let old = State {
-            height: block::Height::from(1001u64),
-            round: 1,
+            height: block::Height::try_from(1001u64).unwrap(),
+            round: block::Round::try_from(1).unwrap(),
             step: 0,
             block_id: None,
         };
 
         let older = State {
-            height: block::Height::from(1001u64),
-            round: 0,
+            height: block::Height::try_from(1001u64).unwrap(),
+            round: block::Round::try_from(0).unwrap(),
             step: 0,
             block_id: None,
         };
 
         let oldest = State {
             height: block::Height::default(),
-            round: 0,
+            round: block::Round::try_from(0).unwrap(),
```

Comment:
LOL, we can change that to `Round::default()` and get on with life. (Maybe in the next PR.)

The longer answer for the try-unwrap dance for constants is this:
We made it a strict requirement that the domain structs we use are safe or we invite abuse of the structs and the possibility of bad code. If constants can add any value without consequences, invalid structs can be created. A Round is a positive value between 0 and i32::MAX. There's no primitive value that can store this in Rust, hence the safe struct that only accepts and emits values between these two. If There's a `Round(pub i32)` definition, then I can create rounds with negative numbers. If there's a `Round(pub u32)` definition, I can create rounds with bigger numbers than i32::MAX. Both create problems when we want to encode to the wire and the current expectation is that encoding to the wire is infallible.

Also, we're only going to deal with constants in our tests. In "real code", we get other variables coming in from the wire. The try-unwrap dance is warranted there. (And it's possibly a `try-?` dance.)

Reply:
Obviously, after my long-winded explanation last night, Romain woke up this morning and asked me something that made me realize how overcomplicated my explanation is. (I always learn something from Romain.)

So, we implemented the From trait for Height in #628 and it will come for the Round struct too with types that will definitely fit into the struct. In the case of Round,  you can do for example: `Round::from(0_u16)`.